designer/client/conditions/SelectConditions.tsx

```diff
@@ -69,6 +77,99 @@ class SelectConditions extends React.Component<Props, State> {
       }, {});
   }
 
+  conditionsForPath(path: string) {
+    const { data } = this.context;
+    const fields: any = Object.values(this.fieldsForPath(path));
+    const { conditions = [] } = data;
+    var conditionsForPath: any[] = [];
+
+    fields.forEach((field) => {
+      const stringConditions = conditions.filter(
+        (condition) => typeof condition.value === "string"
+      );
+      const objectConditions = conditions.filter(
+        (condition) => typeof condition.value !== "string"
+      );
+      this.handleStringConditions(
+        stringConditions,
+        field.name,
+        conditionsForPath
+      );
+      this.handleConditions(objectConditions, field.name, conditionsForPath);
+    });
+
+    return conditionsForPath;
+  }
+
+  handleConditions(
+    objectConditions: ConditionData[],
+    fieldName: string,
+    conditionsForPath: any[]
+  ) {
+    objectConditions.forEach((condition) => {
+      condition.value.conditions.forEach((innerCondition) => {
+        this.checkAndAddCondition(
+          condition,
+          fieldName,
+          innerCondition.field.name,
+          conditionsForPath
+        );
+      });
+    });
+  }
+
+  handleStringConditions(
+    stringConditions: any[],
+    fieldName: string,
+    conditionsForPath: any[]
+  ) {
+    const operators = ["==", "!=", ">", "<"];
+
+    stringConditions.forEach((condition) => {
+      operators.some((operator) => condition.value.includes(operator));
+      if (operators.some((operator) => condition.value.includes(operator))) {
+        var conditionFieldName = condition.value
+          .substring(
+            condition.value.indexOf(".") + 1,
+            condition.value.lastIndexOf(
+              operators.filter((operator) => condition.value.includes(operator))
+            )
+          )
+          .trim();
+        this.checkAndAddCondition(
+          condition,
+          fieldName,
+          conditionFieldName,
+          conditionsForPath
+        );
+      }
```

Comment:
ok - so I think this code is doing

for each string condition, check if any of the operators are used in the current condition, if they are, get the format the condition.value to get conditionFieldName, then check it

For react, it's preferred that you write things in a functional way, so usually this means any arrays/objects you make are immutable. Otherwise, it's hard for react to track what's happening and to update the UI accordingly. and like I mentioned, flatter is easier to read 

so, I would do it like this,
- check if any of the operators are used in the current condition and return a new array containing these (filter) 
  ```
   const conditionsWithAcceptedOperators = stringConditions.filter(condition => operators.some((operator) => condition.value.includes(operator)) 
  ```
- then, I want to change the condition values for each condition in `conditionsWithAcceptedOperators`
  ```
  const conditionsWithFieldName = conditionsWithAcceptedOperators.map(condition => {...condition, conditionFieldName: condition.value.substring(.....)})  
  ``` 

- lastly, I'll iterate over all the conditions I've filtered/updated to check them.
```
  conditionsWithFieldName.forEach(condition => this.checkAndAddCondition(...)) 
``` 
 

now everything can be performed in a flat and performant way

Reply:
So this method should no be flatter now.